src/main/java/in/projecteka/consentmanager/consent/ConsentScheduler.java

```diff
@@ -0,0 +1,160 @@
+                    ConsentRepresentation consentRepresentation = getConsentRepresentation(consentExpiry.getConsentId(), consentExpiry.getPatientId()).block();
+                    if (consentRepresentation != null) {
+                        ConsentRequestDetail consentRequestDetail = consentRequestRepository.requestOf(
+                                consentRepresentation.getConsentRequestId(),
+                                GRANTED.toString(),
+                                consentRepresentation.getConsentDetail().getPatient().getId()).block();
+                        updateAndBroadcastConsentExpiry(
+                                consentExpiry.getConsentId(),
+                                consentRepresentation,
+                                consentRequestDetail);
+                    }
+                }
+            });
+        }
+    }
+
+    private boolean isConsentExpired(Date dateExpiryAt) {
+        Date now = new Date();
+        return (!(dateExpiryAt.after(now) || dateExpiryAt.equals(now)));
+    }
+
+    private Mono<ConsentRepresentation> getConsentRepresentation(String consentId, String requesterId) {
+        return getConsentWithRequest(consentId)
+                .filter(consentRepresentation -> !isNotSameRequester(consentRepresentation.getConsentDetail(), requesterId))
+                .switchIfEmpty(Mono.error(ClientError.consentArtefactForbidden()))
+                .filter(this::isGrantedConsent)
+                .switchIfEmpty(Mono.error(ClientError.consentNotGranted()));
+    }
+
+    private Mono<ConsentRepresentation> getConsentWithRequest(String consentId) {
+        return consentArtefactRepository.getConsentWithRequest(consentId)
+                .switchIfEmpty(Mono.error(ClientError.consentArtefactNotFound()));
+    }
+
+    private static boolean isNotSameRequester(ConsentArtefact consentDetail, String requesterId) {
+        return !consentDetail.getHiu().getId().equals(requesterId) &&
+                !consentDetail.getPatient().getId().equals(requesterId);
+    }
+
+    private boolean isGrantedConsent(ConsentRepresentation consentRepresentation) {
+        return consentRepresentation.getStatus().equals(GRANTED);
+    }
+
+    private void updateAndBroadcastConsentExpiry(String consentId, ConsentRepresentation consentRepresentation, ConsentRequestDetail consentRequestDetail) {
+        consentArtefactRepository.updateConsentArtefactStatus(consentId, EXPIRED).block();
+        var hipConsentArtefactRepresentations = getHIPConsentArtefacts(consentRepresentation);
+        hipConsentArtefactRepresentations.forEach(hipConsentArtefactRepresentation ->
+                broadcastConsentArtefacts(hipConsentArtefactRepresentations,
+                        consentRequestDetail.getConsentNotificationUrl(),
+                        consentRepresentation.getConsentRequestId(),
+                        consentRepresentation.getDateModified()).block());
```

Comment:
Make sure your rows does not exceeds column length / word limit (I think 120 or something).

Reply:
done.